### ISSUE TITLE
External resource causes an error

### ISSUE THREAD

**Issue by mbykovskyy** [marcbachmann/node-html-pdf#71]
When html includes an external resource that takes a bit long to load `html-pdf` throws an error.

```
dev/lp-reports/index.js:11
      throw err;
            ^
Error: html-pdf: Evaluation - TypeError: null is not an object (evaluating 'document.body.outerHTML')
Stack:
  at phantomjs://webpage.evaluate(): 18 (in function )
  at phantomjs://webpage.evaluate(): 26 (in function )

    at ChildProcess.<anonymous> (dev/lp-reports/node_modules/html-pdf/lib/pdf.js:124:17)
    at ChildProcess.emit (events.js:110:17)
    at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
```

I pinned it down to the fact that the content of the page is evaluated before it finishes to load. In `pdf_a4_portrait` script the "Set up content" and "Generate footer & header" parts should move into a `page.onLoadFinished` callback otherwise `body = document.body.outerHTML` inside `page.evaluate` callback may result in a null pointer.

**Comment by mbykovskyy** [comment]
After some more investigation it looks like it only happens when the resource is a javascript.

**Comment by marcbachmann** [comment]
Fixed in v1.4.0
